pyfixest/utils/dev_utils.py

```diff
@@ -176,3 +176,35 @@ def _drop_cols(_data: pd.DataFrame, na_index: np.ndarray):
         return _data[keep]
     else:
         return _data
+
+
+def _extract_variable_level(fe_string: str):
+    """
+    Extract the variable and level from a given string.
+
+    Parameters
+    ----------
+    fe_string: str
+        The string encapsulating the fixed effect factor variable and level.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the extracted variable and level for the fixed
+        effect.
+    """
+    c_pattern = r"C\((.+?)\)"
+    t_pattern = r"\[T\.(.*\])"
+    c_match = re.search(c_pattern, fe_string)
+    t_match = re.search(t_pattern, fe_string, re.DOTALL)
+
+    if not c_match or not t_match:
+        raise ValueError(
+            f"feols() failed after regex encountered the following value as a fixed effect:\n {fe_string}."
+            + "\nThis may due to the presence of line separation and/or escape sequences within the string."
+            + " If so, consider recoding the underlying string. Otherwise, please open a PR in the github repo!"
+        )
+
+    variable = c_match.group(1)
+    level = t_match.group(1)
+    return 'C(' + variable + ')', level[0 : level.rfind("]")]
```

Comment:
Maybe we want to drop the `C(...)` wrapper here? Then we could also drop it further below?

Reply:
Actually, let's do this in a separate PR, as this clashes with Apoorva's PR #553. Likely easier to merge both as is and then adjust =)

Reply:
On this part: `level[0 : level.rfind("]")]` - did you encountered a string where it was needed because the regex did not handle it by itself, or is this just to be on the safe side?

Reply:
The regex matching was leading to values like 'f3' whereas the unit test was expecting 'C(f3)' so this wrapper is necessary to clear the unit test. Similarly,  the initial matching was outputting values such as "['ios'" and "['ios', 'android'" with the rightmost bracket missing. I am not well-versed in regex, but I'm sure there is a more eloquent solution